### PR TITLE
Pass `--fix` option to Ruff in pre-commit

### LINKED ISSUE #1840: Pass `--fix` option to `ruff` in pre-commit
## Problem

`ruff` is able to automatically fix some things like unused imports, except that we aren't passing the `--fix` argument to enable that. eg:

```
catalog/dags/create_filtered_index/dag_factory.py:58:41: F401 [*] `airflow.models.baseoperator.chain` imported but unused
Found 1 error.
[*] 1 potentially fixable with the --fix option.
```

## Description

Update the `ruff` pre-commit hook to pass the `--fix` arg:

```diff
diff --git a/.pre-commit-config.yaml b/.pre-commit-config.yaml
index 5e39c541d..4485487f3 100644
--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -62,6 +62,8 @@ repos:
     rev: "v0.0.261"
     hooks:
       - id: ruff
+        args:
+          - --fix
 
   - repo: https://github.com/ambv/black
     rev: 23.3.0
```

### PR DESCRIPTION
**Pull request by Yasir761**
## Fixes 

Fixes #1840 by @sarayourfriend 

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [ ] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -62,6 +62,8 @@ repos:
     rev: "v0.0.261"
     hooks:
       - id: ruff
+        args:
+          - --fix
```
Comment (dhruvkb) [inline .pre-commit-config.yaml:66]: As per [Ruff's docs](https://beta.ruff.rs/docs/usage/#pre-commit), you will need another flag.
```suggestion
          - --fix
          - --exit-non-zero-on-fix
```

**Review by dhruvkb — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (sarayourfriend) [reply .pre-commit-config.yaml:66]: Not really, pre-commit already fails hooks if any files change.

**Review by sarayourfriend — APPROVED** [review]
LGTM on the basis that pre-commit considers hooks as failed if they change any files.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (dhruvkb) [reply .pre-commit-config.yaml:66]: Ah, I wonder why Ruff wants that flag then.

**Review by dhruvkb — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (sarayourfriend) [reply .pre-commit-config.yaml:66]: I had the same assumption until I tried it myself locally when writing a local hook last week. I think it is a reasonable but incorrect assumption that the Ruff documentation writers have made :slightly_smiling_face:

**Pull request merged**